File: qif4j/src/main/java/com/github/squirrelgrip/qif/write/QifInterestIncomeRecord.java
package com.github.squirrelgrip.qif.write;

import com.github.squirrelgrip.qif.QifInterestIncome;
import com.github.squirrelgrip.qif.QifTransaction;

public class QifInterestIncomeRecord extends AbstractQifRecord {
	private QifInterestIncome txn;
	private QifHeaderEnum header;

	public QifInterestIncomeRecord(QifHeaderEnum header, QifTransaction txn) {
		this.header = header;
		this.txn = (QifInterestIncome) txn;
	}
	
	@Override
	public QifHeaderEnum getHeader() {
		return header;
	}

	@Override
	public String formatTransaction() {
		StringBuilder sb = new StringBuilder();
		Utils.appendFieldValueToOutput("D", txn.getDate(), sb);
		Utils.appendFieldValueToOutput("N", txn.getAction(), sb);
		Utils.appendCurrencyValueToOutput("T", txn.getTotal(), sb);
		return sb.toString();
	}
}

File: qif4k/src/main/kotlin/com/github/squirrelgrip/qif/write/QifInterestIncomeRecord.kt
package com.github.squirrelgrip.qif.write

import com.github.squirrelgrip.qif.QifInterestIncome
import com.github.squirrelgrip.qif.QifTransaction

class QifInterestIncomeRecord(override val header: QifHeaderEnum?, txn: QifTransaction) : AbstractQifRecord() {
    private val txn: QifInterestIncome

    override fun formatTransaction(): String? {
        val sb = StringBuilder()
        Utils.appendFieldValueToOutput("D", txn.date, sb)
        Utils.appendFieldValueToOutput("N", txn.action, sb)
        Utils.appendCurrencyValueToOutput("T", txn.total, sb)
        return sb.toString()
    }

    init {
        this.txn = txn as QifInterestIncome
    }
}